### PR TITLE
UI refactor - abstract

### PR DESCRIPTION
**Pull request by shinyichen**
Depends on #125

### CODE REVIEW

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+export interface ISelectProps {
+  formLabel?: string;
+  options: SelectOption[];
+  defaultOption: string;
+  onOptionSelected: (id: string) => void;
+}
+
+export const Select = ({ formLabel, options, defaultOption, onOptionSelected }: ISelectProps): ReactElement => {
```
Comment (thostetler) [inline src/components/Select/Select.tsx:20]: Why use this custom component instead of Chakra's built-in `Select`? https://chakra-ui.com/docs/form/select

**Review by thostetler — APPROVED** [review]
This LGTM, other than a comment I had.

I also pushed a small commit, I memoized a couple things.  possibly premature optimization though...

File: src/components/Select/Select.tsx (hunk 1 above)
Comment (shinyichen) [reply src/components/Select/Select.tsx:20]: Chakra's select dropdown is native. Here I am using Chakra's menu as dropdown with a customized label.

Comment (shinyichen) [reply src/components/Select/Select.tsx:20]: Looks like in the next PR, I updated this component to use ReactSelect rather than Chakra's menu.

**Pull request merged**